### PR TITLE
Sync locale and lang from WordPress theme

### LINKED ISSUE #199: [Feature] Inherit parent locale in embedded mode
## Problem

When in embedded mode, the Openverse iFrame should communicate with the parent site to set the current i18n locale.

## Description

This work is a companion to WordPress/wordpress.org#39

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes
Fixes #199 by @zackkrida 

## Description

The WordPress theme emits `localeSet` events in response to `localeGet` events (WordPress/wordpress.org#39). This PR sends `localeGet` events and updates the Nuxt i18n locale based on the data of the returned `localeSet` event.

## Technical details

This PR turns off the URL prefix because that leads to the URL containing a language in the subdomain as well as subpath, which might be different leading to inconsistent and bad UX.

## Tests

1. Set up the `wp-openverse` theme locally with `wp-env`.
2. Launch the Nuxt server with `npm run dev`.
3. Set the Nuxt server URL (`http://localhost:8443`) as the `iframe` URL in the theme customizer.

## Screenshots

https://user-images.githubusercontent.com/16580576/132738567-88b387e6-b6c9-4253-ae73-9699b0b3b4d3.mov

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/mixins/i18n-sync.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  data() {
+    return {
+      /**
+       * Handles messages of type `localeSet` received by the `iframe`. Any
+       * other message types will by discarded.
+       *
+       * @param {string} type - the nature of the message received
+       * @param {{ locale: string, lang: string }} value - the data sent with the message
+       */
+      localeMsgHandler: ({ data: { type, value } }) => {
```
Comment (sarayourfriend) [inline src/mixins/i18n-sync.js:14]: Just for my own understanding, why is this function defined in `data` instead of in `methods`?

**Review by sarayourfriend — COMMENTED** [review]
> Set up the wp-openverse theme locally with wp-env

Are there instructions somewhere for how to do this?

**Comment by dhruvkb** [comment]
@sarayourfriend yes, you can see the instructions in the [theme `README.md` file](https://github.com/WordPress/wordpress.org/blob/master/wordpress.org/public_html/wp-content/themes/pub/wporg-openverse/README.md).

File: src/mixins/i18n-sync.js (hunk 1 above)
Comment (dhruvkb) [reply src/mixins/i18n-sync.js]: You're absolutely right about this. It should be a method. Too much composition API has ruined my judgement 😄. Fixed in f8d0728.

**Review by sarayourfriend — APPROVED** [review]
Works great!

**Review by krysal — APPROVED** [review]
So cool to see this working! ⭐️

**Review by zackkrida — APPROVED** [review]

**Review by obulat — APPROVED** [review]
Thank you for adding such detailed testing instructions! So excited for these changes.
I was thinking whether we could replace the mixins with Vue set up functions in the future, is that a feasible idea?

**Comment by dhruvkb** [comment]
@obulat composition API is quite antithetical to mixins so "mixin"g these concepts might not be the best idea. But it'll definitely simplify the code so it deserves some experiementation!

**Pull request merged**